ingest-app/src/cmr/ingest/services/subscriptions_helper.clj

```diff
@@ -201,3 +202,22 @@
                            (remove :deleted)
                            (map #(select-keys % [:concept-id :extra-fields :metadata])))]
     (send-subscription-emails context (process-subscriptions context subscriptions))))
+
+(defn throw-error-if-one-sided-date-map
+  "Throws a service error if the provided map does not contain both keys :start-date and :end-date"
+  [date-map]
+  (when (not-every? date-map [:start-date :end-date])
+    (errors/throw-service-error :bad-request "One-sided date formats are not permitted. Please provide a start and end date."))
+  date-map)
+
+(defn trigger-email-subscription-processing
+  "Trigger subscription processing given the provided revision-date.  Revision date is passed through to
+   a Search query and should follow the same format as the CMR Search API."
+  [context params]
+  (let [revision-date (:revision-date params)]
```

Comment:
nit: I would like to name this parameter something like `revision-date-range` to indicate that it is a range.

Reply:
Yeah, I also found this confusing but stuck with `revision-date` because that is the query parameter it's passed through to in the search query.  We're strictly accepting ranges on this endpoint though, so changing it to `revision-date-range` sounds good to me!